usb/manual/ssdt.md

```diff
@@ -0,0 +1,165 @@
+## SSDT method
+
+A little bit of history worth to mention:
+
+* Thanks to [Gengik84](https://www.macos86.it/profile/1-gengik84/) and the [macos86.it community](https://www.macos86.it/topic/9-mappatura-porte-usb/) for developing and testing this quick and easy method, the original post dates back to 2018, even though some folks [stole it later in 2019](https://www.olarila.com/topic/6181-guide-native-usb-fix-for-desktops-no-injectorkext-required-skylake/) without mentioning any credits.
```

Comment:
This method of mapping has existed for years, I'm not really sure about attributing all the credit to Gengik84/macos86.it for it (although make no mistake, I'm not trying to diminish their contributions or anything). The stolen remark is also useless and will just lead to more traffic, there is no point in including it.

Reply:
The "*quick and easy method*" I was talking about in the guide referred to the `Method` itself described in the `.txt` - not the port naming itself, such as 0x03 for external USB 3.0 or 0x00 for 2.0 - which was defined in ACPI spec

> This method of mapping has existed for years

I cannot find the `GENG` method anywhere else except in those websites who stole it, please post where this method was reported before the creator wrote it

> will just lead to more traffic

This is true, but 

> The stolen remark is also useless

this was my way to make it clear that others often steal what you do and they sell it as their own effort - that's very sick with open source. For the scope of the guide it is indeed useless, but developers are people too and not robots, hence it was worth on my end to mention why the guide came so much later from its creation in 2018

Reply:
Mapping through ACPI has existed for years:
https://www.insanelymac.com/forum/topic/253502-dsdt-fix-for-usb2/
https://www.insanelymac.com/forum/topic/323540-thunderbolt-drivers/
https://www.insanelymac.com/forum/topic/215986-kind-of-dsdt-fix-for-usb-device-dead-after-wake-from-sleep-on-1063/
https://www.insanelymac.com/forum/topic/334203-ssdt-high-sierra-native-usb-on-ryzen-without-genericusbxhci-or-injector-kexts/

Mapping through ACPI is not new, nor using a method to reduce repetition in _UPC/_PLD.

I get how you feel about having the guide stolen, but for the end-user all that will happen is that they will go to the other guide and potentially use it.

Reply:
Maybe I wasn't clear enough...
I'm talking about the single `Method` to map all the ports, not the _UPC/PLD

The method GENG was stolen by Maldon and renamed to MALD in here:
https://www.olarila.com/topic/6181-guide-native-usb-fix-for-desktops-no-injectorkext-required-skylake/

Reply:
By the way I removed the maldon part in the guide. Now the credits are there for the  unique`Method` only to define both the ports and the connectors - which has never existed, even in the links you forwarded

Reply:
The source for the `GENG` method is the motherboard ACPI writers themselves, who have sourced it from Intel's internal dev kits. For example, said `GENG` method is identical to one found in my DSDT:
```
Method (UUPC, 2, Serialized)
{
    Name (PCKG, Package (0x04)
    {
        Zero, 
        0xFF, 
        Zero, 
        Zero
    })
    PCKG [Zero] = Arg0
    PCKG [One] = Arg1
    Return (PCKG) /* \_SB_.PCI0.XHC_.RHUB.UUPC.PCKG */
}
```
Vendors have been doing this for a while, this is not new by any means.

Reply:
> Vendors have been doing this for a while, this is not new by any means.

I could not find any firmware made before the beginning of 2018 implementing this, please provide more examples where that method was present even previously

Reply:
But after all, are there any issues in mentioning someone's work in the credits?

Reply:
> I could not find any firmware made before the beginning of 2018 implementing this, please provide more examples where that method was present even previously

@1alessandro1 Both my MSI Z170A and my HP Elite X2 have this method used for their RHUB devices, which are both Skylake series devices. Again, looking even at the Intel Skylake developer kit leaks, we can see this is even used by intel themselves